python/datafusion/context.py

```diff
@@ -0,0 +1,1167 @@
+        self.config_internal = self.config_internal.with_disk_manager_specified(paths)
+        return self
+
+    def with_unbounded_memory_pool(self) -> RuntimeConfig:
+        """Use an unbounded memory pool.
+
+        Returns
+        -------
+        RuntimeConfig
+            A new `RuntimeConfig` object with the updated setting.
+
+        Examples
+        --------
+        >>> config = RuntimeConfig().with_unbounded_memory_pool()
+        """
+        self.config_internal = self.config_internal.with_unbounded_memory_pool()
+        return self
+
+    def with_fair_spill_pool(self, size: int) -> RuntimeConfig:
+        """Use a fair spill pool with the specified size.
+
+        This pool works best when you know beforehand the query has multiple spillable
+        operators that will likely all need to spill. Sometimes it will cause spills
+        even when there was sufficient memory (reserved for other operators) to avoid
+        doing so.
+
+        ```text
+            ┌───────────────────────z──────────────────────z───────────────┐
+            │                       z                      z               │
+            │                       z                      z               │
+            │       Spillable       z       Unspillable    z     Free      │
+            │        Memory         z        Memory        z    Memory     │
+            │                       z                      z               │
+            │                       z                      z               │
+            └───────────────────────z──────────────────────z───────────────┘
+        ```
+
+        Parameters
+        ----------
+        size : int
+            Size of the memory pool in bytes.
+
+        Returns
+        -------
+        RuntimeConfig
+            A new `RuntimeConfig` object with the updated setting.
+
+        Examples
+        --------
+        ```python
+        >>> config = RuntimeConfig().with_fair_spill_pool(1024)
+        ```
+        """
+        self.config_internal = self.config_internal.with_fair_spill_pool(size)
+        return self
+
+    def with_greedy_memory_pool(self, size: int) -> RuntimeConfig:
+        """Use a greedy memory pool with the specified size.
+
+        This pool works well for queries that do not need to spill or have a single
+        spillable operator. See `RuntimeConfig.with_fair_spill_pool` if there are
+        multiple spillable operators that all will spill.
+
+        Parameters
+        ----------
+        size : int
+            Size of the memory pool in bytes.
+
+        Returns
+        -------
+        RuntimeConfig
+            A new `RuntimeConfig` object with the updated setting.
+
+        Examples
+        --------
+        >>> config = RuntimeConfig().with_greedy_memory_pool(1024)
+        """
+        self.config_internal = self.config_internal.with_greedy_memory_pool(size)
+        return self
+
+    def with_temp_file_path(self, path: str) -> RuntimeConfig:
```

Comment:
Can we support `pathlib.Path` for this?

Reply:
Added, but not resolving this comment yet until unit tests are also added